File: basic/autograd/cpp/tensor_test.cpp
#include "torch/script.h"
#include <iostream>

int main()
{
	// Create a tensorand set requires_grad = True to track computation with it
	torch::Tensor x = torch::ones({ 2, 2 }, torch::requires_grad());
	std::cout << x << std::endl;

	// Do a tensor operation
	torch::Tensor y = x + 2;
	std::cout << y << std::endl;

	// change requires_grad flag in - place
	torch::Tensor a = torch::randn({ 2, 2 });
	std::cout << a.grad_fn() << std::endl;
	a = ((a * 3) / (a - 1));
	std::cout << a.grad_fn() << std::endl;
	std::cout << a.requires_grad() << std::endl;
	a.requires_grad_(true);
	std::cout << a.requires_grad() << std::endl;
	a = ((a * 3) / (a - 1));
	std::cout << a.grad_fn() << std::endl;

	return 0;
}

File: basic/autograd/python/tensor_test.py
# -*- coding: utf-8 -*-
import torch

# Create a tensor and set requires_grad=True to track computation with it
x = torch.ones(2, 2, requires_grad=True)
print(x)

# Do a tensor operation
y = x + 2
print(y)

# change requires_grad flag in-place
a = torch.randn(2, 2)
print(a.grad_fn)
a = ((a * 3) / (a - 1))
print(a.grad_fn)
print(a.requires_grad)
a.requires_grad_(True)
print(a.requires_grad)
a = ((a * 3) / (a - 1))
print(a.grad_fn)
